chirp/drivers/icm710.py

```diff
@@ -0,0 +1,227 @@
+        rf.has_name = True
+        rf.has_bank = False
+        rf.has_bank_names = False
+        rf.has_tuning_step = False
+        rf.has_ctone = False
+        rf.has_cross = True
+        rf.has_infinite_number = False
+        rf.has_nostep_tuning = True
+        rf.has_comment = True
+        rf.has_settings = False  #TODO radio does have settings
+        rf.has_variable_power = False
+        
+        rf.memory_bounds = (0, 232)
+        rf.valid_bands = [(1000000, 29999999)] #[(1000000, 166666665)] 
+        
+        rf.valid_duplexes = DUPLEX_MODES
+        rf.valid_modes = MODULATION_MODES          
+        rf.valid_power_levels = POWER_LEVELS
+        rf.valid_tones = []                        
+        rf.valid_tmodes = []
+        rf.valid_skips = []
+        #rf.valid_tuning_steps = [1.0, 2.5, 5.0]        
+        return rf
+
+    # Convert the raw byte array into a memory object structure
+    def process_mmap(self):
+        self._memobj = bitwise.parse(MEM_FORMAT, self._mmap)
+
+    # Return a raw representation of the memory object, which
+    # is very helpful for development
+    def get_raw_memory(self, number):
+        return repr(self._memobj.memory[number])
+
+    def sync_out(self):
+        self.pipe.baudrate = 4800
+        icf.clone_to_radio(self)
+        
+    # Extract a high-level memory object from the low-level memory map
+    # This is called to populate a memory in the UI
+    def get_memory(self, number):
+        
+        # this just gets the 160 common 
+        # the 72 ITU channels are probably working
+        # there may be some strange behavior with the SITOR FSK "hidden" channels
+        # best be careful about trying to set a fixed simplex channel with a duplex frequency
+        # TODO test, test, test, all the combinations. 
+        
+        # Get a low-level memory object mapped to the image
+        _mem = self._memobj.memory[ (int(number)/2) ]
+
+        # Create a high-level memory object to return to the UI
+        mem = chirp_common.Memory()
+
+        mem.number = number                 # Set the memory number
+        mem.comment = ''
+
+        # Convert your low-level frequency to Hertz		_mem.freq	Exception occurred during evaluation.	
+        if (number) % 2 == 0:
+            mem.freq = int(_mem.rx_freqL)
+            mem.name = str(_mem.nameL).rstrip()  # Set the alpha tag
+            mem.mode = MODULATION_MODES[int(_mem.modeL)]
+            #mem.power = POWER_LEVELS[int(2)]     # TODO, have to find where this is in the record!
+            mem.offset = int(_mem.tx_freqL)      # if the frequency is above 30MHz (166666665 or FF,FF,FF,FF), it's simplex
+            if int(_mem.tx_freqL) < 30000000:
+                 mem.duplex = 'split'
+                 mem.comment = 'TX: ' + str((int(_mem.tx_freqL)/1000000)) + ' MHz'
+            else:
+                mem.duplex = 'off'      
+                mem.comment = ''
+                _mem.tx_freqL.set_raw('\xFF\xFF\xFF\xFF')
```

Comment:
There is lots of leftover bad practice from the python2 days in the tree, but please pass `bytes` to `set_raw()` and not strings.

Reply:
Done